src/main/java/org/springframework/samples/petclinic/model/NamedEntity.java

```diff
@@ -45,4 +46,17 @@ public String toString() {
         return this.getName();
     }
 
+    @Override
```

Comment:
Are these new methods actually used anywhere?

Reply:
@dsyer the updated test relies indirectly on `equals`. IMO, I am with you that the implementation isn't good enough to show best practices so I am rather keen to decline this. 

Having said that, the current test that this PR changes is problematic so we should do something about it.